src/app/profile/user-profile/user-profile.component.html

```
@@ -131,15 +181,23 @@ <h2 class="title-medium mb-2">Discount Cards</h2>
                             (click)="refreshEsnCard()"
                             [disabled]="refreshCardMutation.isPending()"
                           >
-                            Refresh
+                            @if (refreshCardMutation.isPending()) {
+                              Refreshing...
+                            } @else {
+                              Refresh
+                            }
                           </button>
                           <button
                             mat-stroked-button
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)